src/Pact/Native/Db.hs

```diff
@@ -354,12 +364,16 @@ guardTable i t = evalError' i $ "Internal error: guardTable called with non-tabl
 
 enforceBlessedHashes :: FunApp -> ModuleName -> Hash -> Eval e ()
 enforceBlessedHashes i mn h = do
-  mmRs <- fmap fst . HM.lookup mn <$> view (eeRefStore.rsModules)
+  mmRs <- fmap _mdModule . HM.lookup mn <$> view (eeRefStore . rsModules) -- fmap fst . HM.lookup mn <$> view (eeRefStore.rsModules)
   mm <- maybe (HM.lookup mn <$> use (evalRefs.rsLoadedModules)) (return.Just) mmRs
-  case mm of
+  case mm of 
     Nothing -> evalError' i $ "Internal error: Module " ++ show mn ++ " not found, could not enforce hashes"
-    Just Module{..}
-      | h == _mHash -> return () -- current version ok
-      | h `HS.member` _mBlessed -> return () -- hash is blessed
-      | otherwise -> evalError' i $
-                     "Execution aborted, hash not blessed for module " ++ show mn ++ ": " ++ show h
+    Just m ->
+      case m of
+        Module{..}
+          | h == _mHash -> return () -- current version ok
+          | h `HS.member` _mBlessed -> return () -- hash is blessed
+          | otherwise -> evalError' i $
+            "Execution aborted, hash not blessed for module " ++ show mn ++ ": " ++ show h
+        _ -> evalError' i $
+          "Execution aborted, hashes not supported for interfaces " ++ show mn ++ ": " ++ show h 
```

Comment:
Not accurate error, it's actually an internal error (we should never have a table declared in a module so this code should never be called)

Reply:
Do you mean tables should never be declared in _interfaces_?